packages/webapp/src/components/SensorReadingsLineChart/index.jsx

```diff
@@ -25,12 +26,62 @@ const PureSensorReadingsLineChart = ({
   yAxisLabel,
   chartData,
 }) => {
-  const [selectedLine, setSelectedLine] = useState(null);
+  const [legendsList, setLegendsList] = useState({});
 
-  const selectLine = (event) => {
-    if (event.dataKey.includes(NO_DATA)) return;
-    let sl = selectedLine === event.dataKey ? null : event.dataKey.trim();
-    setSelectedLine(sl);
+  useEffect(() => {
+    if (yAxisDataKeys.length) {
+      setLegendsList(
+        yAxisDataKeys.reduce((acc, cv, idx) => {
```

Comment:
We can probably get rid of `idx` as it doesn't look like it is being used in the reduce loop

Reply:
I am using it on line 38.

Reply:
:+1:

Reply:
Ah my bad, I missed that